### ISSUE TITLE
Bad Request (400) returned from Elasticsearch Server (v7.4.2)

### ISSUE THREAD

**Issue by pfeigl** [bruno-garcia/log4net.ElasticSearch#120]
We are seeing the following error for any log request which we make towards our elasticsearch instance

```
Warning: 299 Elasticsearch-7.4.2-2f90bbf7b93631e52bafb59b3b049cb44ec25e96 "[types removal] Specifying types in document index requests is deprecated, use the typeless endpoints instead (/{index}/_doc/{id}, /{index}/_doc, or /{index}/_create/{id})."
```

Are we doing anything wrong? We tried the latest version of this library `2.6.0-preview2`

Thanks,
Philipp

**Comment by pfeigl** [comment]
Possibly an important deatil: For testing I set the bufferSize to "1" to have every logline transferred to Elasticsearch, so I was probably testing without the bulk API.

**Comment by bruno-garcia** [comment]
Does this happen with the bulk API? IMHO the non-bulk API doesn't make much sense in any real production environment so if that's the culprit, maybe nothing really to be done except perhaps dropping support.
Also, can you please test it with `2.5.0`? Maybe it's a problem introduced in the new version which is preview.

**Comment by pfeigl** [comment]
I tried 2.5.0 before updating to the preview versions - with the same effect.
The problem only comes up when using the bulk API, I just changed the bufferSize to 1 to see every log coming in instantly. I have to agree, that this is probably not a production usecase.

I just had problems with the bulk API at first too and thought it would be the same problem, but later found out, that I had to name the index all lowercase.

What made all of this complicated for me was the fact, that I couldn't see the logging errors even while setting all the log4net debug settings and tracelog to enabled.
The errors were only visible when I used fiddler to track the problems down.
